### ISSUE TITLE
Sufficiently large negative float parses to int incorrectly.

### ISSUE THREAD

**Issue by tbremer** [facebook/reason#2291]
This could be a mistake on my part, but when running a large negative float through `int_of_float` they start inverting and parse positive.

```reason
let flt = -1.4249758164e+13;
Js.log(int_of_float(flt)); /* results: 943324128 */
```

the above example should parse to: `-14249758164000`

Online REPL:
https://reasonml.github.io/en/try?rrjsx=true&reason=DYUwLgBAZskLwQLQEYB0AWATOgnAdgFYAOZANnRAGpkBmAbgCgApAZ1WAHsBzACgEsAdmAD6HKMJgcAhmB4wwASgVA

**Comment by OvermindDL1** [comment]
This seems correct to me as the natural integer value in OCaml is 31 bits, and in javascript it is 32 bits, and that number well *well* exceeds both of those, requiring 48 bits of storage (which is getting close to the max of a `double` as well that javascript itself uses for the integral representation).  Thus the number is getting wrapped or truncated or something since it is significantly out of range.

**Comment by tbremer** [comment]
Yes, all of that makes sense. It's rather large and odd number to be computing, I can give some context as to how I got here.

I was parsing date strings from the 1500s (advent of code) and came across (what I thought to be) anomalies on how the times were being printed output.

So, the real use was along these lines:

```reason
let a = [|"1518-06-12T00:00",
"1518-10-08T00:19",
"1518-07-21T00:53"|];

let to_int = i => i
	->Js.Date.fromString
	->Js.Date.getTime
	->int_of_float;

Js.log(Array.map(to_int, a));
/* [922288128,-1766273760,100832] */
```

The solution was to skip the `to_int` step, however I was surprised by the (perceived) inconsistencies in the results.

**Comment by jaredly** [comment]
I don't think there's anything actionable here :/ for reference, the variable `max_int` in reason indicates how high/low you can safely go with integers, and it's `2,147,483,647`.
